src/oncall/user_sync/ldap_sync.py

```diff
@@ -382,12 +384,19 @@ def sync(config, engine):
     for username in users_to_purge:
         prune_user(engine, username)
 
+    # mark active and re-add contact information
     logger.debug('Users to reactivate:')
     for username in users_to_reactivate:
         logger.debug(username)
         try:
             engine.execute('UPDATE user SET active = TRUE WHERE name = %s', username)
             stats['users_reactivated'] += 1
+            for key, value in ldap_users[username].items():
+                if value and key in modes:
+                    logger.debug('\t%s -> %s' % (key, value))
+                    user_contact_add_sql = 'INSERT INTO `user_contact` (`user_id`, `mode_id`, `destination`) VALUES (%s, %s, %s)'
+                    engine.execute(user_contact_add_sql, (user_id, modes[key], value))
```

Comment:
I might be missing something but where is the `user_id` variable defined? I can't find it. Should the user id interpolation should be converted to a subquery that takes in the username instead?

Reply:
You're not missing anything, I absentmindedly pushed an intermediate set of changes. I've updated the PR with the full changes.